### PR TITLE
Modcc/from to block

### PR DESCRIPTION
**Pull request by thorstenhater**
Add parsing for FROM ... TO ... descriptions in parser.

### CODE REVIEW

File: modcc/parser.cpp [hunk 1]
```diff
@@ -782,6 +786,31 @@ std::pair<Token, Token> Parser::range_description() {
...
+std::pair<Token, Token> Parser::from_to_description() {
+    Token lb, ub;
+
+    if(token_.type != tok::from) {
+        error(pprintf("range description must start with a FROM '%'", token_));
+        return {};
+    }
+
+    get_token();
+    lb = token_;
```
Comment (bcumming) [inline modcc/parser.cpp:798]: Test that `token_` is a value?

File: modcc/parser.cpp [hunk 2]
```diff
@@ -399,9 +399,13 @@ void Parser::parse_state_block() {
         }
 
         parm.token = token_;
-        //state_block.state_variables.push_back(token_.spelling);
         get_token();
 
+        if(token_.type == tok::from) {
```
Comment (bcumming) [inline modcc/parser.cpp:404]: Should this check for an incorrectly formed `FROM <value> To <value>` statement?
If an error was set, it should return, I think.

File: modcc/parser.cpp (hunk 1 above)
Comment (thorstenhater) [reply modcc/parser.cpp:798]: Good idea. We should add this to <lo, hi> ranges as well.

File: modcc/parser.cpp (hunk 2 above)
Comment (bcumming) [reply modcc/parser.cpp:404]: This part is right, and you should call `from_to_description()` should call `error` if it detects an error in the statement.
You can then check if an error was set, and just `return`.

File: modcc/parser.cpp [hunk 3]
```diff
@@ -399,9 +399,17 @@ void Parser::parse_state_block() {
...
+            // silently skips from/to
+            from_to_description();
+            if (status_ == lexerStatus::error) {
+                error(pprintf("range description must be of form FROM <int> TO <int>: '%'", token_));
```
Comment (bcumming) [inline modcc/parser.cpp:408]: I don't think you need to set the error message here: it was already set in the call to `from_to_description()`, right?

File: modcc/parser.cpp [hunk 4]
```diff
@@ -820,6 +869,14 @@ expression_ptr Parser::parse_prototype(std::string name=std::string()) {
 
         get_token(); // consume the identifier
 
+        // args may have a unit attached
+        if(token_.type == tok::lparen) {
+            unit_description();
+            if(status_ == lexerStatus::error) {
```
Comment (bcumming) [inline modcc/parser.cpp:875]: `unit_description` has already set the error message, so we can just return.

**Comment by thorstenhater** [comment]
Done in #1050

**Pull request closed without merging**